### PR TITLE
ContainerLaunchPriority support container launch timeout alert

### PR DESCRIPTION
**Pull request by ls-2018**
- https://github.com/openkruise/kruise/issues/1422

### CODE REVIEW

**Comment by ls-2018** [comment]
@zmberg Can you look at this request again please? Thanks.

File: pkg/controller/containerlaunchpriority/container_launch_priority_controller.go [hunk 1]
```diff
@@ -151,23 +158,74 @@ func (r *ReconcileContainerLaunchPriority) Reconcile(_ context.Context, request
...
+			parse, err := time.Parse(time.RFC3339, updateStr)
+			if err == nil {
+				updateTime = parse
+			}
+		}
+		containerStatus := util.GetContainerStatus(container.Name, pod)
+
+		if timeout > 0 && time.Duration(timeout)*time.Second < time.Since(updateTime) && (containerStatus == nil || containerStatus.Ready == false) {
+			name, _ := os.Hostname()
+			event := &v1.Event{
```
Comment (zmberg) [inline pkg/controller/containerlaunchpriority/container_launch_priority_controller.go:181]: You can call p.recorder.Eventf() function to record event

**Pull request closed without merging**